Shared/Extensions/JellyfinAPIExtensions/BaseItemDto+VideoPlayerViewModel.swift

```diff
@@ -166,4 +166,159 @@ extension BaseItemDto {
+					                                                                                 requireAvc: true,
+					                                                                                 transcodingMaxAudioChannels: 6,
+					                                                                                 videoCodec: videoStream?.codec,
+					                                                                                 videoStreamIndex: videoStream?.index,
+					                                                                                 enableAdaptiveBitrateStreaming: true)
+
+					var hlsStreamComponents = URLComponents(string: hlsStreamBuilder.URLString)!
+					hlsStreamComponents.addQueryItem(name: "api_key", value: SessionManager.main.currentLogin.user.accessToken)
+
+					hlsStreamURL = hlsStreamComponents.url!
+
+					// MARK: VidoPlayerViewModel Creation
+
+					var subtitle: String?
+
+					// MARK: Attach media content to self
+
+					var modifiedSelfItem = self
+					modifiedSelfItem.mediaStreams = currentMediaSource.mediaStreams
+
+					// TODO: other forms of media subtitle
+					if self.itemType == .episode {
+						if let seriesName = self.seriesName, let episodeLocator = self.getEpisodeLocator() {
+							subtitle = "\(seriesName) - \(episodeLocator)"
+						}
+					}
+
+					let subtitlesEnabled = defaultSubtitleStream != nil
+
+					let shouldShowAutoPlay = Defaults[.shouldShowAutoPlay] && itemType == .episode
+					let autoplayEnabled = Defaults[.autoplayEnabled] && shouldShowAutoPlay
+
+					let overlayType = Defaults[.overlayType]
+
+					let shouldShowPlayPreviousItem = Defaults[.shouldShowPlayPreviousItem] && itemType == .episode
+					let shouldShowPlayNextItem = Defaults[.shouldShowPlayNextItem] && itemType == .episode
+
+					var fileName: String?
+					if let lastInPath = currentMediaSource.path?.split(separator: "/").last {
+						fileName = String(lastInPath)
+					}
+
+					let videoPlayerViewModel = VideoPlayerViewModel(item: modifiedSelfItem,
+					                                                title: modifiedSelfItem.name ?? "",
+					                                                subtitle: subtitle,
+					                                                directStreamURL: directStreamURL,
+					                                                transcodedStreamURL: transcodedStreamURL?.url,
+					                                                hlsStreamURL: hlsStreamURL,
+					                                                streamType: streamType,
+					                                                response: response,
+					                                                audioStreams: audioStreams,
+					                                                subtitleStreams: subtitleStreams,
+					                                                chapters: modifiedSelfItem.chapters ?? [],
+					                                                selectedAudioStreamIndex: defaultAudioStream?.index ?? -1,
```

Comment:
This is where the audio stream index gets set and then is set in the media player. I assume that the channel stream doesn't have a default audio stream.

Reply:
If I put a breakpoint on 294 and take a look at this at the moment I select a liveTV channel item for playback, I am actually finding that the defaultAudioStream is set, to a stereo aac audio track. Also this track appears to have an index value of `-1`, which is how it matches up on line 202 where the default stream is selected from the audio streams array. In this case, that array also only has this same one audio stream object. 

![Screen Shot 2022-01-24 at 6 14 38 PM](https://user-images.githubusercontent.com/125045/150886648-f7db5949-fa13-457f-a409-f06e813875e9.png)

Reply:
Interesting, as I don't know how live tv objects are stored in JF in respect to a `BaseItemDto` object. Over time, you'll have to do more investigation and tailor the video players to the data from these objects.